### ISSUE TITLE
How to measure data fetching, forward and backward pass time during training

### ISSUE THREAD

**Issue by PurvangL** [tensorflow/tensorflow#61904]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1.  It must be a bug, a feature request, or a significant problem with the
    documentation (for small docs fixes please send a PR instead).
2.  The form below must be filled out.
3.  It shouldn't be a TensorBoard issue. Those go
    [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**:   No
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**:
-   **TensorFlow installed from (source or binary)**: Binary
-   **TensorFlow version (use command below)**: 2.11.0
-   **Python version**: 3.8
-   **Bazel version (if compiling from source)**:
-   **GCC/Compiler version (if compiling from source)**:
-   **CUDA/cuDNN version**: 12
-   **GPU model and memory**: A100
-   **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
How to measure Data fetching, Data preparation, forward and backward pass time for training (script)[https://www.tensorflow.org/tutorials/images/segmentation]?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by sushreebarsa** [comment]
@PurvangL Generally you can get the time in the model.fit logs which will be taken/step. Could you please check this [guide](https://www.tensorflow.org/guide/autodiff) for more details on the gradients differentiation and how that is calculated during the training. It would have been easier to understand your query if any sample code was provided. I could also see there is one ongoing thread on TF [Forum](https://discuss.tensorflow.org/t/how-to-measure-data-fetching-forward-and-backward-pass-time-during-training/19698). As this issue is more related to support so please close this ticket and we will follow up in the other thread. Thank you!

**Comment by PurvangL** [comment]
Thanks @sushreebarsa. Closing issue here.
